tests/test_qnode.py

```diff
@@ -109,6 +110,36 @@ def operable_mock_device_2_wires():
         yield Device(wires=2)
 
 
+class TestQNodeConstruct:
+    """
+    Tests methods that are called during construction of QNode
+    """
+    def test_best_method_with_non_gaussian_successors(self):
+        dev = DummyDevice(wires=2)
+
+        @qml.qnode(dev)
+        def circuit(x):
+            qml.Squeezing(x, 0, wires=[0])
+            qml.Beamsplitter(np.pi/4, 0, wires=[0, 1])
+            qml.Kerr(0.54, wires=[1])
+            return qml.expval(qml.NumberOperator(0))
+
+        circuit.jacobian([0.321], method='A')
```

Comment:
This needs to be updated to compare the two methods:
```suggestion
        res = circuit.jacobian([0.321], method='A')
        expected = circuit.jacobian([0.321], method='F')
        assert np.allclose(res, expected, atol=tol, rtol=0)
```